website/integrations: kanboard (#14515)
goauthentik/authentik: website/integrations/services/kanboard/index.md

:::

## authentik configuration

To support the integration of Kanboard with authentik, you need to create an application/provider pair in authentik.

### Create an application and provider in authentik

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.
- **Choose a Provider type**: select **OAuth2/OpenID Connect** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Note the **Client ID** and **Client Secret** values because they will be required later.
    - Set a `Strict` redirect URI to `https://kanboard.company/oauth/callback`.
    - Select any available signing key.

> - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

Reviewer: Can you indent these lines please

3. Click **Submit** to save the new application and provider.

## Kanboard configuration

Integrating Kanboard with authentik requires enabling the plugin system and installing the OAuth2 plugin.

### Enable plugin management

To enable plugin management through the web interface, add the following line to your Kanboard configuration file, typically located at `/var/www/app/config.php`:

> ```yaml showLineNumbers

Reviewer: ```suggestion
```yaml
```

define('PLUGIN_INSTALLER', true);
```

> Once that is done, restart your server to apply the updated configuration.

Reviewer: ```suggestion
Then restart your server to apply the updated configuration.
```

### Install and configure the plugin

> 

Reviewer: Do we need a mention of "Login to your Kanban admin dashboard" or similar here?

1. Install the OAuth2 plugin by navigating to your **Profile Icon** > **Settings** > **Plugins**.
2. Locate the **OAuth2** plugin in the list and click **Install**.

> 3. After installation, navigate to **Profile Icon** > **Settings** > **Integrations**.

Reviewer: ```suggestion
3. After the installation is complete, navigate to **Profile Icon** > **Settings** > **Integrations**.
```

| -------------------------- | ----------------------------------------------------- |
| **Callback URL**           | `https://kanboard.company/oauth/callback` (prefilled) |
| **Client ID**              | Client ID from authentik                              |
| **Client Secret**          | Client secret from authentik                          |
| **Authorize URL**          | `https://authentik.company/application/o/authorize`   |
| **Token URL**              | `https://authentik.company/application/o/token`       |
| **User API URL**           | `https://authentik.company/application/o/userinfo`    |
| **Scopes**                 | `openid profile email`                                |
| **Username Key**           | `preferred_username`                                  |
| **Name Key**               | `name`                                                |
| **Email Key**              | `email`                                               |
| **User ID Key**            | `sub`                                                 |
| **Allow Account Creation** | Toggled                                               |

5. Click **Save** to apply your configuration.

> ## Resources

Reviewer: Can you move the resources section to the end
